uefi/src/fs/path/mod.rs

```diff
@@ -0,0 +1,53 @@
+//! This module offers the [`Path`] and [`PathBuf`] abstractions.
+//!
+//! # Interoperability with Rust strings
+//!
+//! For the interoperability with Rust strings, i.e., `String` and `str` from
+//! the standard library, the API is intended to transform these types first to
+//! `CString16` respectively `CStr16`. They do not directly translate to
+//! [`Path`] and [`PathBuf`].
+//!
+//! # Path Structure
+//!
+//! Paths use the [`SEPARATOR`] character as separator. Paths are absolute and
+//! do not contain `.` or `..` components. However, this can be implemented in
+//! the future.
+
+mod path;
+mod pathbuf;
+mod validation;
+
+pub use path::Path;
```

Comment:
I think we should also expose `Components` here so that it gets linked in the rustdoc ouput for the `Path::components()` method.

Reply:
good catch